dtfs-central-api/api-tests/helpers/with-validate-audit-details.api-tests.ts

```diff
@@ -0,0 +1,52 @@
+import { AuditDetails } from '@ukef/dtfs2-common/src/types/audit-details';
+import {
+  generatePortalAuditDetails,
+  generateSystemAuditDetails,
+  generateTfmAuditDetails,
+} from '@ukef/dtfs2-common/src/helpers/change-stream/generate-audit-details';
+import { MOCK_TFM_USER } from '../mocks/test-users/mock-tfm-user';
+import { MOCK_PORTAL_USER } from '../mocks/test-users/mock-portal-user';
+
+type Params = {
+  makeRequest: (auditDetails?: AuditDetails) => Promise<{ status: number; body: object }>;
+  validUserTypes: ('tfm' | 'portal' | 'system')[];
+};
+
+export const withValidateAuditDetailsTests = ({ makeRequest, validUserTypes }: Params) => {
+  describe('when validating audit details', () => {
+    it('should return 400 if no auditDetails provided', async () => {
+      const { status, body } = await makeRequest();
+
+      expect(body).toEqual({ status: 400, message: 'Invalid auditDetails, Missing property `userType`' });
+      expect(status).toBe(400);
+    });
+
+    const testCases = [
+      {
+        auditDetails: generateSystemAuditDetails(),
+        expectedStatus: validUserTypes.includes('system') ? 200 : 400,
+      },
+      {
+        auditDetails: generatePortalAuditDetails(MOCK_PORTAL_USER._id),
+        expectedStatus: validUserTypes.includes('portal') ? 200 : 400,
+      },
+      {
+        auditDetails: generateTfmAuditDetails(MOCK_TFM_USER._id),
+        expectedStatus: validUserTypes.includes('tfm') ? 200 : 400,
+      },
+    ];
+
+    it.each(testCases)('it should have status $expectedStatus if the userType is $auditDetails.usertype', async ({ auditDetails, expectedStatus }) => {
```

Comment:
Note to self
`$auditDetails.usertype` doesnt work - not sure why

Getting
'it should have status 400 if the userType is {"id": "6602f646141e3d545a91fbd1", "userType": "portal"}'

Reply:
very very odd. Seems to imply it should work in the docs... I don't mind if you want to just create a `description` value or something and assign `userType` to this...
```
Or generate unique test titles by injecting properties of test case object with $variable
To inject nested object values use you can supply a keyPath i.e. $variable.path.to.value
```

https://jestjs.io/docs/api